### ISSUE TITLE
Switch from Python pickles to other serialization format

### ISSUE THREAD

**Issue by michwill** [zero-db/zerodb#8]
Since we could have native clients in languages other than Python, we'd like to switch to object serialization other than pickle. Something similar to jsonpickle + msgpack (but without [arbitrary code execution](http://versprite.com/og/into-the-jar-jsonpickle-exploitation/)). This is a longer term goal

**Comment by minsler** [comment]
:+1:

**Comment by HarryR** [comment]
http://jmoiron.net/blog/python-serialization/

Ultimately for compatibility JSON encoding rather than pickling seems like the natural choice, the problem with pickle is that it's inherently tied Python.

**Comment by michwill** [comment]
Certainly. I'm thinking of something more compact and fast, yet cross-language. What about msgpack?

**Comment by IAlwaysBeCoding** [comment]
how about [thrift](https://diwakergupta.github.io/thrift-missing-guide/)?

**Comment by michwill** [comment]
Fastest in Python seems to be msgpack. We will first switch to it in ZEO protocol, then in zerodb itself
